File: CvManagementSystem/CvManagementSystem.Api/Contracts/Positions/CreateUpdatePositionRequest.cs
using UserService.Api.Contracts.Projects;
using UserService.Domain.Models;

namespace UserService.Api.Contracts.Positions;

public class CreateUpdatePositionRequest
{
    public string Title { get; set; } = null!;
    public string Description { get; set; } = null!;
    public ExpertiseLevel ExpertiseLevel { get; set; }
    public List<CreateUpdateAccessRuleRequest> AccessRules { get; set; } = null!;
    public List<CreateUpdateTechnologyRequest> Technologies { get; set; } = null!;
    public uint MaxProjects { get; set; }
    public bool Restricted { get; set; }
    public uint Version { get; set; }
}

File: Frontend/src/models/CreateUpdatePositionRequest.ts
import {Technology} from "./Technology";
import {ExpertiseLevel} from "./ExpertiseLevel";
import {CreateUpdateAccessRuleRequest} from "./CreateUpdateAccessRuleRequest";

export interface CreateUpdatePositionRequest {
    title: string;
    description: string;
    expertiseLevel: ExpertiseLevel;
    accessRules: CreateUpdateAccessRuleRequest[];
    technologies: Technology[];
    maxProjects: number;
    restricted: boolean;
    version: number;
}